diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,8 +25,13 @@ endif()
 
 # Find Eigen3 to build the library
-find_package(Eigen3 REQUIRED NO_MODULE)
-if(Eigen3_FOUND)
-    message(STATUS "Found Eigen3")
-    message(STATUS "Eigen3 include dir: ${EIGEN3_INCLUDE_DIRS}")
+if(EIGEN3_INCLUDE_DIRS)
+    set(Eigen3_FOUND TRUE)
+    message(STATUS "Using provided Eigen3 include dir: ${EIGEN3_INCLUDE_DIRS}")
 else()
-    message(FATAL_ERROR "Could not locate Eigen3")
+    find_package(Eigen3 REQUIRED NO_MODULE)
+    if(Eigen3_FOUND)
+        message(STATUS "Found Eigen3")
+        message(STATUS "Eigen3 include dir: ${EIGEN3_INCLUDE_DIRS}")
+    else()
+        message(FATAL_ERROR "Could not locate Eigen3")
+    endif()
 endif()
 
 # Find OSQP library and headers
